# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/colo/make_fix-1.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Submitted By: Joe Ciccone <joeciccone@crazyeyesoft.com>
Date: 2005-11-23
Initial Package Version: 1.19
Upstream Status: Sent
Origin: Joe Ciccone <joeciccone@crazyeyesoft.com
Description: Adds flags to the stage1, stage2, and chain Makefiles
             forcing them to build as 32bit.

	     2-2-06 Updated for Colo 1.21 
diff -Naur colo-1.21.orig/chain/Makefile colo-1.21/chain/Makefile
--- colo-1.21.orig/chain/Makefile	2006-01-17 23:03:58.000000000 +0000
+++ colo-1.21/chain/Makefile	2006-02-02 15:43:32.000000000 +0000
@@ -18,9 +18,9 @@
 
 DEPS=$(OBJS:.o=.d)
 
-LDFLAGS= -nostdlib -Wl,-T,$(TARG).lnk,-Map,$(TARG).map
-CFLAGS= -Werror -Wall -Wstrict-prototypes -fomit-frame-pointer -O2 -fno-strict-aliasing $(CFLAGS_CPU) $(CFLAGS_COLO)
-CPPFLAGS= -MD -nostdinc -Iinclude -I../include -D_DEBUG $(CPPFLAGS_GCC)
+LDFLAGS= -mabi=32 -nostdlib -Wl,-T,$(TARG).lnk,-Map,$(TARG).map
+CFLAGS= -mabi=32 -Werror -Wall -Wstrict-prototypes -fomit-frame-pointer -O2 -fno-strict-aliasing $(CFLAGS_CPU) $(CFLAGS_COLO)
+CPPFLAGS= -mabi=32 -MD -nostdinc -Iinclude -I../include -D_DEBUG $(CPPFLAGS_GCC)
 
 all:
 	$(MAKE) -C .. binary
diff -Naur colo-1.21.orig/stage1/Makefile colo-1.21/stage1/Makefile
--- colo-1.21.orig/stage1/Makefile	2006-01-17 23:03:58.000000000 +0000
+++ colo-1.21/stage1/Makefile	2006-02-02 15:44:18.000000000 +0000
@@ -21,9 +21,9 @@
 
 DEPS=$(OBJS:.o=.d)
 
-LDFLAGS= -nostdlib -Wl,-T,$(TARG).lnk,-Map,$(TARG).map
-CFLAGS= -Werror -Wall -Wstrict-prototypes -fomit-frame-pointer -O2 -fno-strict-aliasing $(CFLAGS_CPU) $(CFLAGS_COLO)
-CPPFLAGS= -MD -nostdinc -Iinclude -I../include -D_DEBUG $(CPPFLAGS_GCC)
+LDFLAGS= -mabi=32 -nostdlib -Wl,-T,$(TARG).lnk,-Map,$(TARG).map
+CFLAGS= -mabi=32 -Werror -Wall -Wstrict-prototypes -fomit-frame-pointer -O2 -fno-strict-aliasing $(CFLAGS_CPU) $(CFLAGS_COLO)
+CPPFLAGS= -mabi=32 -MD -nostdinc -Iinclude -I../include -D_DEBUG $(CPPFLAGS_GCC)
 
 all:
 	$(MAKE) -C .. binary
diff -Naur colo-1.21.orig/stage2/Makefile colo-1.21/stage2/Makefile
--- colo-1.21.orig/stage2/Makefile	2006-01-17 23:03:58.000000000 +0000
+++ colo-1.21/stage2/Makefile	2006-02-02 15:44:02.000000000 +0000
@@ -54,9 +54,9 @@
 
 DEPS=$(OBJS:.o=.d)
 
-LDFLAGS= -nostdlib -Wl,-T,$(TARG).lnk,-Map,$(TARG).map,--emit-relocs
-CFLAGS= -Werror -Wall -Wstrict-prototypes -fomit-frame-pointer -O2 -fno-strict-aliasing $(CFLAGS_CPU) $(CFLAGS_COLO)
-CPPFLAGS= -MD -nostdinc -Iinclude -I../include -D_DEBUG $(CPPFLAGS_GCC)
+LDFLAGS= -mabi=32 -nostdlib -Wl,-T,$(TARG).lnk,-Map,$(TARG).map,--emit-relocs
+CFLAGS= -mabi=32 -Werror -Wall -Wstrict-prototypes -fomit-frame-pointer -O2 -fno-strict-aliasing $(CFLAGS_CPU) $(CFLAGS_COLO)
+CPPFLAGS= -mabi=32 -MD -nostdinc -Iinclude -I../include -D_DEBUG $(CPPFLAGS_GCC)
 
 LOADADDR= $(shell $(CROSS_COMPILE)readelf -l $(TARG) | awk '$$1 ~ /^LOAD$$/ { print $$3 }')
 
